pkg/providers/aws/cloudwatch.go

```diff
@@ -29,22 +30,26 @@ func NewCloudWatchClient(cfg *aws.Config) *cloudWatchClient {
 		return nil
 	}
 
-	// Return the ec2 service
+	// Return the cloudwatch service client
 	return &cloudWatchClient{
 		client: client,
 	}
 }
 
 // Get the resource consumption of an ec2 instance
-func (e *cloudWatchClient) GetEc2Metrics(region awsRegion, cache *awsCache) map[string]v1.Instance {
+func (e *cloudWatchClient) GetEC2Metrics(region awsRegion, cache *awsCache) (map[string]v1.Instance, error) {
 	serviceMetrics := make(map[string]v1.Instance)
 
 	// Define the period
+	// TODO: This should use the window
 	end := time.Now().UTC()
 	start := end.Add(-5 * time.Minute)
 
 	// Get the cpu consumption for all the instances in the region
-	if cpuMetrics := e.getEc2Cpu(region, start, end); len(cpuMetrics) > 0 {
+	if cpuMetrics, err := e.getEC2CPU(region, start, end); len(cpuMetrics) > 0 {
+		if err != nil {
+			return serviceMetrics, err
+		}
 		for _, cpuMetric := range cpuMetrics {
```

Comment:
once we add memory we will need to to the same instance lookup dance that was implemented in the GCP provider